### PR TITLE
Allow component to update with new props when a button is pressed

### PR DESCRIPTION
**Pull request by eeeli24**

### CODE REVIEW

**Review by iusfof — APPROVED** [review]

File: MultiSlider.js [hunk 1]
```diff
@@ -1,534 +1,455 @@
...
+                nextState.valueTwo = nextProps.values[1];
+                nextState.positionTwo = valueToPosition(nextProps.values[1], this.optionsArray, nextProps.sliderLength);
+                const positionTwoShift = nextState.positionTwo - this.state.positionTwo;
+                nextState.pastTwo = this.state.twoPressed
+                    ? this.state.pastTwo + positionTwoShift
+                    : nextState.positionTwo;
+            }
+        }
+
+        if (nextState != {}) {
```
Comment (basvasilich) [inline MultiSlider.js]: `_.isEmpty(nextState)` 
`{} != {}` always true

File: MultiSlider.js [hunk 2]
```diff
@@ -1,534 +1,455 @@
...
+
+    static defaultProps = {
+        values: [0],
+        onValuesChangeStart: () => {},
+        onValuesChange: values => values,
+        onValuesChangeFinish: values => values,
+        step: 1,
+        min: 0,
+        max: 10,
+        touchDimensions: {
```
Comment (basvasilich) [inline MultiSlider.js:62]: `styles`

File: MultiSlider.js [hunk 3]
```diff
@@ -1,534 +1,455 @@
...
+                nextState.pastTwo = this.state.twoPressed
+                    ? this.state.pastTwo + positionTwoShift
+                    : nextState.positionTwo;
+            }
+        }
+
+        if (nextState != {}) {
+            this.setState(nextState);
+        }
     }
```
Comment (basvasilich) [inline MultiSlider.js]: Need some comments here about what's going on

File: MultiSlider.js [hunk 4]
```diff
@@ -1,534 +1,455 @@
...
+                        let change = [this.state.valueOne];
+                        if (this.state.valueTwo) {
+                            change.push(this.state.valueTwo);
+                        }
+                        this.props.onValuesChange(change);
+                    }
+                );
+            }
+        }
+    };
```
Comment (basvasilich) [inline MultiSlider.js:220]: Need some comments here about what's going on

File: MultiSlider.js [hunk 5]
```diff
@@ -1,534 +1,455 @@
...
+                    {
+                        valueTwo: value
+                    },
+                    () => {
+                        this.props.onValuesChange([this.state.valueOne, this.state.valueTwo]);
+                    }
+                );
+            }
+        }
+    };
```
Comment (basvasilich) [inline MultiSlider.js:255]: Need some comments here about what's going on

File: MultiSlider.js (hunk 2 above)
Comment (eeeli24) [reply MultiSlider.js:62]: Those aren't really styles, `slipDisplacement` is used to calculate position for example

File: MultiSlider.js (hunk 4 above)
Comment (eeeli24) [reply MultiSlider.js:220]: added

File: MultiSlider.js (hunk 5 above)
Comment (eeeli24) [reply MultiSlider.js:255]: added for positionOne(pretty much the same here for positionTwo)

**Review by sunr1ze — APPROVED** [review]

**Review by basvasilich — APPROVED** [review]

**Pull request merged**